### ISSUE TITLE
SqlClient Managed SNI tries to establish a UDP connection for Sql Server connections

### ISSUE THREAD

**Issue by saurabh500** [dotnet/corefx#15089]
SqlClient Managed SNI profiling shows that there is a significant amount of time being spent to open a connection while resolving the default instance port `SNIProxy.TryTOGetDefaultInstancePort(...)`

https://github.com/dotnet/corefx/blob/bffef76f6af208e2042a2f27bc081ee908bb390b/src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNIProxy.cs#L267

This may be unnecessary for most use cases. 
I am opening this issue to find a way to optimize this. 

![image](https://cloud.githubusercontent.com/assets/1623701/21869018/676231b2-d80a-11e6-9d2d-ce586653c95d.png)

cc @corivera @geleems

**Comment by saurabh500** [comment]
A work around is to specify the port of the SqlServer so that the UDP connection is not made to find the port number.

**Comment by saurabh500** [comment]
I did some testing with SqlClient on .Net Framework.

Server setup:
1. For the MSSQLServer instance (which is the default instance) the only protocol turned on was Tcp, SM and NP were disabled.
2. I changed the TCP port of Sql server from 1433 to 1432 so that the default port is not picked up during connectivity. 
3. The connection to sql Server with "Data Source = servername" (Note no port, no instance name fails) 

If I enabled Named Pipes / Shared memory on the Sql Server and try to connect using above data source, the connection succeeds. That's likely because of named pipes connecting to the default pipe which is enabled for the Sql Server.

The resolution is to remove the SSRP lookup when no port or instance is specified in case of a TCP connection. This would significantly improve the connectivity in case there is no need to get to the SqlBrowser process.
